osu.Game.Rulesets.Taiko/Skinning/Legacy/LegacySwell.cs

```diff
@@ -0,0 +1,136 @@
+
+                        Children = new Drawable[]
+                        {
+                            spinnerCircle = new Sprite
+                            {
+                                Texture = skin.GetTexture("spinner-circle"),
+                                Anchor = Anchor.Centre,
+                                Origin = Anchor.Centre,
+                                Scale = new Vector2(0.8f),
+                            },
+                            shrinkingRing = new Sprite
+                            {
+                                Texture = skin.GetTexture("spinner-approachcircle") ?? skinManager.DefaultClassicSkin.GetTexture("spinner-approachcircle"),
+                                Anchor = Anchor.Centre,
+                                Origin = Anchor.Centre,
+                                Scale = Vector2.One,
+                            },
+                            remainingHitsCountdown = new LegacySpriteText(LegacyFont.Combo)
+                            {
+                                Anchor = Anchor.Centre,
+                                Origin = Anchor.Centre,
+                                Position = new Vector2(0f, 165f),
+                                Scale = Vector2.One,
+                            },
+                        }
+                    },
+                    clearAnimation = new Sprite
+                    {
+                        // File extension is included here because of a GetTexture limitation, see #21543
+                        Texture = skin.GetTexture("spinner-osu.png"),
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                        Position = new Vector2(0f, -165f),
+                        Scale = new Vector2(0.3f),
+                        Alpha = 0,
+                    },
+                }
+            };
+
+            clearSample = skin.GetSample(new SampleInfo("spinner-osu"));
+        }
+
+        public void AnimateSwellProgress(DrawableTaikoHitObject<Swell> swell, int numHits, SkinnableDrawable mainPiece)
+        {
+            remainingHitsCountdown.Text = $"{swell.HitObject.RequiredHits - numHits}";
+            spinnerCircle.RotateTo(180f * numHits, 1000, Easing.OutQuint);
+        }
+
+        public void AnimateSwellCompletion(ArmedState state, SkinnableDrawable mainPiece)
+        {
+            const double clear_transition_duration = 300;
+
+            bodyContainer.FadeOut(clear_transition_duration, Easing.OutQuad);
+
+            if (state == ArmedState.Hit)
+            {
+                if (!samplePlayed)
+                {
+                    clearSample?.Play();
+                    samplePlayed = true;
+                }
+
+                clearAnimation
+                    .FadeIn(clear_transition_duration, Easing.InQuad)
+                    .ScaleTo(0.8f, clear_transition_duration, Easing.InQuad)
+                    .Delay(700).FadeOut(200, Easing.OutQuad);
```

Comment:
Seems slightly off. Stable source says:

```csharp
            p.Transformations.Add(
                new Transformation(new Vector2(320, 240), new Vector2(320, 150), AudioEngine.Time, (int)(AudioEngine.Time + HitFadeIn * 2), EasingTypes.Out));
            p.Transformations.Add(
                new Transformation(TransformationType.Scale, 0.4F, 1F, AudioEngine.Time,
                                   (int)(AudioEngine.Time + (HitFadeIn * 2)), EasingTypes.Out));
            p.Transformations.Add(
                new Transformation(TransformationType.Fade, 1, 0,
                                   AudioEngine.Time + HitFadeIn * 3,
                                   (int)(AudioEngine.Time + (HitFadeIn * 5.5))));
```

`HitFadeIn` is 120 (constant). Easing type is presumed `None` if not specified explicitly.